src/components/NodeSelectorDialog.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  chainStorageWatcherAtom,
+  isNodeSelectorOpenAtom,
+  networkConfigAtom,
+  rpcNodeAtom,
+  savedApiNodeAtom,
+  savedRpcNodeAtom,
+} from 'store/app';
+import { useAtom, useAtomValue, useSetAtom } from 'jotai';
+import ActionsDialog from './ActionsDialog';
+import Combobox from 'components/Combobox';
+import { useEffect, useRef, useState } from 'react';
+import { fetchAllAddrs } from 'utils/rpc';
+
+const useRpcAddrs = () => {
+  const [rpcAddrs, setRpcAddrs] = useState([]);
+  const [apiAddrs, setApiAddrs] = useState([]);
+  const networkConfig = useAtomValue(networkConfigAtom);
+
+  useEffect(() => {
+    const fetchAddrs = async () => {
+      const { rpcAddrs, apiAddrs } = await fetchAllAddrs(networkConfig.url);
+      setRpcAddrs(rpcAddrs);
+      setApiAddrs(apiAddrs);
+    };
+
+    try {
+      fetchAddrs();
+    } catch (e) {
+      console.error('Error loading RPC Addrs', e);
+    }
+  }, [networkConfig]);
+
+  return { rpcAddrs, apiAddrs };
+};
+const NodeSelectorDialog = () => {
+  const [isOpen, setIsOpen] = useAtom(isNodeSelectorOpenAtom);
+  const { rpcAddrs, apiAddrs } = useRpcAddrs();
+  const watcher = useAtomValue(chainStorageWatcherAtom);
+  const rpcAddr = useAtomValue(rpcNodeAtom);
+  const [api, setApi] = useState(watcher?.apiAddr);
+  const [rpc, setRpc] = useState(rpcAddr ?? undefined);
+  const [initialApi, setInitialApi] = useState(api);
+  const [initialRpc, setInitialRpc] = useState(rpc);
+  const setSavedRpc = useSetAtom(savedRpcNodeAtom);
+  const setSavedApi = useSetAtom(savedApiNodeAtom);
+
+  const save = () => {
+    assert(api && rpc);
+    setSavedApi(api);
+    setSavedRpc(rpc);
+    setIsOpen(false);
+    window.location.reload();
```

Comment:
Is a full page reload necessary here? I guess it doesn't matter much, as a majority of app state is dependent on the rpc + api endpoints.

Reply:
Yea it's just taking the easy way out. Otherwise we'd have to recreate the chainStorageWatcher and walletConnection, reset the state, and make sure we clean up all casting followers etc.